src/Pact/Types/Term.hs

```diff
@@ -1166,6 +1166,18 @@ tLit = (`TLiteral` def)
 tStr :: Text -> Term n
 tStr = toTerm
 
+-- | Equality dictionary for term-level equality
+--
+canEq :: Term n -> Term n -> Bool
+canEq t u = case (t,u) of
```

Comment:
Before anyone comments, the `TList{}` syntax only works for records/ADTs with nonstrict fields.

Reply:
:(

Reply:
Fixed - the previous comment applies to pattern guards only. I can explicitly case on `{}` and it does what i need.